lib/sparse/src/index/inverted_index/inverted_index_ram.rs

```diff
@@ -21,31 +20,12 @@ pub struct InvertedIndexRam {
 }
 
 impl InvertedIndex for InvertedIndexRam {
-    fn open(path: &Path) -> std::io::Result<Self> {
-        let mmap_inverted_index = InvertedIndexMmap::load(path)?;
-        let mut inverted_index = InvertedIndexRam {
-            postings: Default::default(),
-            vector_count: mmap_inverted_index.file_header.vector_count,
-        };
-
-        for i in 0..mmap_inverted_index.file_header.posting_count as DimId {
-            let posting_list = mmap_inverted_index.get(&i).ok_or_else(|| {
-                std::io::Error::new(
-                    std::io::ErrorKind::InvalidData,
-                    format!("Posting list {} not found", i),
-                )
-            })?;
-            inverted_index.postings.push(PostingList {
-                elements: posting_list.to_owned(),
-            });
-        }
-
-        Ok(inverted_index)
+    fn open(_path: &Path) -> std::io::Result<Self> {
+        panic!("InvertedIndexRam is not supposed to be loaded");
```

Comment:
Prior to this PR, `InvertedIndexRam` was used both for mutable and immutable cases.

For the mutable case, it was never loaded:
https://github.com/qdrant/qdrant/blob/c230a48c50d90c32a118a645d67b2df7d1e35375/lib/segment/src/index/sparse_index/sparse_vector_index.rs#L95-L105

And never saved:
https://github.com/qdrant/qdrant/blob/c230a48c50d90c32a118a645d67b2df7d1e35375/lib/segment/src/index/sparse_index/sparse_vector_index.rs#L492-L496

And immutable case was not supporting upsert:
https://github.com/qdrant/qdrant/blob/c230a48c50d90c32a118a645d67b2df7d1e35375/lib/segment/src/index/sparse_index/sparse_vector_index.rs#L530-L534

---

In this PR I made this distinction more clear:
- `InvertedIndexRam`: Mutable case, supports upserts, but no load/save.
- `InvertedIndexImmutableRam`: Immutable case, supports load/save, but no upserts.

Reply:
In theory, I don't see a problem why Mutable index should not be loaded. But in our scenarios it is, indeed, never used

Reply:
I agree that load&save is invalid here. My question is about design. Can you redesign `InvertedIndex` and remove `save` and `load` functions to avoid unnecessary panic here?
I propose to do the same as vector storage: constructor is a separate function wich know stoarge/index type, ref:
https://github.com/qdrant/qdrant/blob/dev/lib/segment/src/vector_storage/dense/simple_dense_vector_storage.rs#L41